doc/build-macos.md

```diff
@@ -3,56 +3,91 @@ macOS Build Instructions and Notes
 The commands in this guide should be executed in a Terminal application.
 The built-in one is located in `/Applications/Utilities/Terminal.app`.
 
-Preparation
------------
-Install the macOS command line tools:
-
-`xcode-select --install`
-
-When the popup appears, click `Install`.
-
-Then install [Homebrew](http://brew.sh).
-
-Dependencies
-----------------------
-
-    brew install automake berkeley-db4 libtool boost miniupnpc openssl pkg-config protobuf python qt libevent qrencode
-
-In case you want to build the disk image with `make deploy` (.dmg / optional), you need RSVG
-
-      brew install librsvg
-      
-Berkley DB
-------------------------
+## Preparation
+1. Install macOS Command Line Tools (if not already installed):
+   ```bash
+   xcode-select --install
+   ```
+   When the popup appears, click `Install`.
+
+
+2. Install Homebrew (if not already installed):
+   ```bash
+   /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+   ```
+
+
+## Dependencies
+Install the required dependencies using Homebrew:
+```bash
+brew install automake berkeley-db4 libtool boost miniupnpc openssl pkg-config protobuf python qt libevent qrencode python-setuptools m4
+```
+
+In case you want to build the disk image with `make deploy` (.dmg / optional), you need RSVG:
+```bash
+brew install librsvg
+```
+### Ensure `m4` is Found
+After installing `m4`, ensure that it is correctly linked and available in your PATH:
+```bash
+brew link m4
+```
```

Comment:
This returned a warning to pass `--force` to the command to properly set symlinks.